enumerables.rb

```diff
@@ -0,0 +1,114 @@
+module Enumerable
+  def my_each()
+    i = 0
+    while i < size
+      return to_enum unless block_given?
+
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      return to_enum unless block_given?
+
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    selected_array = []
+    i = 0
+    while i < size
+      return to_enum unless block_given?
+
+      selected_array << self[i] if yield(self[i])
+      i += 1
+    end
+    selected_array
+  end
+
+  def my_all?
+    i = 0
+    while i < size
+      return false if block_given? && !yield(self[i])
+      return false if self[i] == false || self[i].nil?
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?
+    i = 0
+    while i < size
+      return true if block_given? && yield(self[i])
+      return true if !self[i] == false && !self[i].nil?
+
+      i += 1
+    end
+    false
+  end
+
+  def my_none?
+    i = 0
+    while i < size
+      if block_given?
+        return false if yield(self[i])
+      elsif !self[i] == false && !self[i].nil?
+        return false
+      end
+
+      i += 1
+    end
+    true
+  end
+
+  def my_count(item = nil)
+    return size if !block_given? && item.nil?
+
+    count = 0
+    my_each do |n|
+      count += 1 if (block_given? && yield(n)) || (n == item)
+    end
+    count
+  end
+
+  def my_map(&proc)
+    return to_enum unless block_given?
+
+    array = to_a
+    mapped_array = []
+    i = 0
+
+    while i < array.size
+      mapped_array << proc.call(array[i])
+      i += 1
+    end
+    mapped_array
+  end
+
+  def my_inject(param1 = nil, param2 = nil)
+    array = to_a
+    accumulator = array[0]
+    i = 1
+
+    accumulator = yield(param1, accumulator) if block_given? && !param1.nil?
+    while i < array.size
+      accumulator = if block_given?
+                      yield(accumulator, array[i])
+                    elsif param2.nil?
+                      accumulator.send(param1, array[i])
+                    else
+                      accumulator.send(param2, array[i]).send(param2, param1)
+                    end
+      i += 1
+    end
+    accumulator
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method
```ruby
array = Array.new(100) { rand(0...9) }
array.my_inject(:+) == array.inject(:+) #true
```

Reply:
`array = Array.new(100) { rand(0...9) }`
`array.my_inject(:+) == array.inject(:+) #true`
The code above returns to true in my machine without any changes done after code review.